### DISCUSSION TITLE
Volume Control Issue with SoundSource and MonitorControl

### DISCUSSION THREAD

**Discussion by ghost** [MonitorControl/MonitorControl#1559] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Looked through [the wiki](https://github.com/MonitorControl/MonitorControl/wiki)
- [X] Updated MonitorControl to the latest version (if applicable)

#### Rogue Amoeba SoundSource Compatibility Issue Disrupting Volume Control

After installing SoundSource, the volume control feature in MonitorControl stopped working correctly. Previously, adjusting volume via keyboard keys was based on the mouse pointer's location—over the MacBook Pro for its volume and over the monitor for the monitor's volume. However, since adding SoundSource, this pointer-dependent volume control no longer functions.

### Steps to reproduce

1. Install SoundSource on a system already using MonitorControl.
2. Attempt to adjust volume using keyboard keys with the mouse pointer over the MacBook Pro or an external monitor.

### Expected behavior

Volume adjusts based on the mouse pointer's location, affecting either the MacBook Pro or the monitor's volume.

### Actual behavior 

Volume control does not respond to the mouse pointer's location, and no adjustment occurs.

### Anything else?

This functionality worked flawlessly before the SoundSource installation. The issue appears to be a compatibility problem between SoundSource and MonitorControl. The affected feature is crucial for users needing seamless volume control across multiple devices.
Even after quitting SoundSwitch, the functionality to control Audio devices is stopped completely from monitor control.

#### Suggested Resolution:
Please investigate potential conflicts between SoundSource and MonitorControl to restore the intuitive volume control feature. Integration of this functionality or a compatibility fix would be greatly appreciated.

Link to MonitorControl: [MonitorControl GitHub](https://github.com/MonitorControl/MonitorControl)
Link to SoundSource: [Rogue Amoeba's SoundSource](https://rogueamoeba.com/soundsource/)

### Environment Information (please complete the following information)

- Mac model: MacBook Pro 16" M1 Pro 1TB
- Apple Silicon/M1 (yes or no): Yes
- MonitorControl Version: 4.2.0 Build 7048
- Operating System: macOS Sonoma 14.4.1

- SoundSource Version: 5.6.3
- App Version: 5.6.3
- Build ID: 04dce93e03cf3c4
- Build Date: 2023-10-19 16:19:14
- ACE Version: 11.9.6

![SoundSource disrupts MonitorControl's volume adjustments for MacBook Pro and monitor keyboard functions.](https://github.com/MonitorControl/MonitorControl/assets/165494359/d364e88f-906e-48c2-b983-2c08cce84baa)

**Comment by ghost** [comment]
# Mouse-Position-Dependent Volume Control

Feature Request: Mouse Location-Based Volume Control

**Title:**
Adaptive Volume Control Based on Mouse Position for Multiple Displays

**Description:**
I propose a feature for MonitorControl that enables volume adjustments according to the mouse's position on different displays. Currently, the volume keys affect only the system's default audio device, which can be inconvenient for users with multiple monitors.

**Use Case:**
For individuals using a MacBook Pro in conjunction with external monitors, the ability to control each monitor's volume independently, based on where the mouse is located, would significantly enhance productivity and ease of use.

**Expected Functionality:**
- When the mouse is on the MacBook Pro's display, volume keys should adjust its volume.
- When the mouse is on an external monitor, volume keys should adjust that monitor's volume.

This feature would facilitate a more intuitive interaction with audio controls in multi-monitor setups, improving MonitorControl's overall user experience.

**Comment by waydabber** [comment]
Sadly we do not accept new feature requests to MonitorControl as the team decided that the app is already too complex compared to its original vision. I plan to add volume control for the built-in display (or rather the built-in speakers of the MBP) for consistency's sake in [BetterDisplay](https://github.com/waydabber/BetterDisplay).

https://github.com/waydabber/BetterDisplay/issues/2314
